mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/host/CassandraRestartExtension.java

```diff
@@ -0,0 +1,38 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.host;
+
+import org.apache.james.backends.cassandra.DockerCassandraSingleton;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class CassandraRestartExtension implements BeforeAllCallback, BeforeEachCallback {
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        DockerCassandraSingleton.restartAfterMaxTestsPlayed();
```

Comment:
Why it's not in beforeEach?

Reply:
Tried at first, it is not working. It seems (if I understood well) that if you try to restart the container after creating and starting the cluster, it crashes the rest of your test suite